README.md

```diff
@@ -75,9 +75,7 @@ jobs:
       - name: Create Release Pull Request or Publish to npm
         uses: changesets/action@master
         with:
-          # The --otp=1 is to get around a bug in changesets
-          # It will be fixed in Changesets 2
-          publish: yarn release --otp=1
+          publish: yarn changeset publish
```

Comment:
The original reasoning behind the publish option was to allow building before publish but this made me think more, I'm thinking this option should be removed and publish should happen if an NPM_TOKEN is provided and a build option should be added that's run before publish. Any thoughts on this?

Reply:
So was the idea it would run `yarn release` which could be their own aliased yarn command doing whatever? But it assumes that they added `changeset publish` somewhere in the release because we pass in the otp?

Agree that it should run a build thing. I wouldn't mind explicitly setting a publish boolean (or setting a publish command being the trigger for this) to make when the bot will publish more explicit than implicit.

Also, since we don't need to pass anything to the changeset command, it might be worth having it run `yarn pre-publish` and then the bot run the changeset command? It cuts the bot off from random publish scripts in non-changeset repos but otherwise seems fine.

Reply:
> setting a publish command being the trigger for this

checked code to confirm this is what it does, will document loudly that the existence of this triggers this.

Reply:
> So was the idea it would run yarn release which could be their own aliased yarn command doing whatever? But it assumes that they added changeset publish somewhere in the release because we pass in the otp?

Yes, that's correct. I know it's bad, it was a "this is a bad idea but I want this to work and doing this will make it work" thing.

> Agree that it should run a build thing. I wouldn't mind explicitly setting a publish boolean (or setting a publish command being the trigger for this) to make when the bot will publish more explicit than implicit.

SGTM

> Also, since we don't need to pass anything to the changeset command, it might be worth having it run yarn pre-publish and then the bot run the changeset command? It cuts the bot off from random publish scripts in non-changeset repos but otherwise seems fine.

As in make this action run a pre-publish script or the Changesets CLI?